### DISCUSSION TITLE
Nuxt 3 - How to use vite plugins?

### DISCUSSION THREAD

**Discussion by scolastico** [nuxt/nuxt#15903] [Questions] [answered]
Hi, i would like to use a vite plugin (https://www.npmjs.com/package/vite-plugin-singlefile) but i dont see anything in the docs mentioning how to load vite plugins with nuxt.

Update:
~~I think i found the config option but it doesnt seem to load. But this could also be an issue of the plugin.~~

**Comment by scolastico** [comment, marked as answer]
Found the option:

```ts
import { defineNuxtConfig } from 'nuxt'
import { viteSingleFile } from "vite-plugin-singlefile"

// https://v3.nuxtjs.org/api/configuration/nuxt.config
export default defineNuxtConfig({

    vite: {
        plugins: [
            viteSingleFile()
        ]
    }

})
```

**Reply by santiagoarizti** [reply]
how do you achieve this same thing through nuxt plugins? the thing is that you cannot make a vite plugin run after a nuxt plugin, so it fails (e.g. cannot load vuetify plugin before vue plugin)

**Reply by ThomasBerne** [reply]
Same question as @santiagoarizti.

**Reply by scolastico** [reply]
@santiagoarizti @ThomasBerne I see 2 possibilities without diving into this topic:

a) write a wrapper for the nuxt configuration which then injects itself as a nuxt plugin and the needed vite plugin into the nuxt configuration, e.g.:
```js
import { defineNuxtConfig } from 'nuxt'.
import myPlugin from 'myPlugin'.

export default defineNuxtConfig(myPlugin({
  // nuxt config  
}))
```

or
b) Just document that it is necessary to install the vite plugin in the nuxt config as well.

**Reply by matthew-dean** [reply]
One thing I'm discovering is that setting Vite plugins at all essentially breaks Nuxt, or at least breaks whatever Vite plugins were already added by Nuxt plugins. 🤷‍♂️

**Reply by matthew-dean** [reply]
Oh... maybe this is the way? https://nuxt.com/docs/api/kit/builder
